airbyte-ci/connectors/pipelines/pipelines/dagger/actions/secrets.py

```diff
@@ -136,21 +136,34 @@ async def mounted_connector_secrets(context: PipelineContext, secret_directory_p
         # Special case for local development.
         # Query dagger for the contents of the secrets and mount these strings as files in the container.
         contents = {}
-        for secret_file_name, secret in context.connector_secrets.items():
-            contents[secret_file_name] = await secret.plaintext()
+        if secret_directory_path:
+            for secret_file_name, secret in context.connector_secrets.items():
+                contents[secret_file_name] = await secret.plaintext()
 
         def with_secrets_mounted_as_regular_files(container: Container) -> Container:
-            container = container.with_exec(["mkdir", "-p", secret_directory_path], skip_entrypoint=True)
-            for secret_file_name, secret_content_str in contents.items():
-                container = container.with_new_file(f"{secret_directory_path}/{secret_file_name}", secret_content_str, permissions=0o600)
+            if context.s3_build_cache_access_key_id:
+                container = container.with_env_variable("S3_BUILD_CACHE_ACCESS_KEY_ID", context.s3_build_cache_access_key_id)
+                if context.s3_build_cache_secret_key:
+                    container = container.with_env_variable("S3_BUILD_CACHE_SECRET_KEY", context.s3_build_cache_secret_key)
```

Comment:
I'd prefer these env var to be explicitly set from the `GradleTask` step as it is Gradle specific.
So far `mounted_connector_secrets` has been used to mount secret connector configuration we're hosting on GSM.

Reply:
The only reason I didn't do so is to not have to duplicate the commentary. Also at the time of writing this code I assumed I'd be running airbyte-ci with those credentials locally, but that hasn't happened, nor was it necessary. I'll rewrite this.

Reply:
Fixed.